src/main/java/md/miller1995/Dealership/services/users/UserAuthDetailsServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package md.miller1995.Dealership.services.users;
+
+import md.miller1995.Dealership.models.entities.UserEntity;
+import md.miller1995.Dealership.repositories.UserRepository;
+import md.miller1995.Dealership.securities.UserAuthDetails;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class UserAuthDetailsServiceImpl implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public UserAuthDetailsServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<UserEntity> user = userRepository.findUserEntityByUsername(username);
+        if(user.isEmpty())
+            throw new UsernameNotFoundException("user not found!");
```

Comment:
Even if there is a single line of code under an `if` statement, for better visibility I would use `{ }`.
Why this method throws UsernameNotFoundException if we handle this exception?

Reply:
- I used brackets for single line statement;
- I will create a handle exception for all exception.